src/components/header/Header.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { useSelectedTheme, useSetSelectedApi, useSetSelectedTheme, useAvailableApis } from '../../state/CharacterProvider';
+
+const Header = () => {
+  const setSelectedTheme = useSetSelectedTheme();
+  const selectedTheme = useSelectedTheme();
+  const setSelectedApi = useSetSelectedApi();
+  const availableApis = useAvailableApis();
+
+  const handleInputChange = ({ target }) => {
+    if(target.checked) {
+      setSelectedTheme('dark');
+    } else setSelectedTheme('light');
+  };
+  
+  const handleSelectChange = ({ target }) => {
+    setSelectedApi(target.value);
+  }
+
+  return (
+    <header className={ 
+      selectedTheme === 'light' 
+        ? header + lightHeader
+        : header + darkHeader
+      }>
+      <div className={wrap}>
+        <div className="relative inline-flex">
+          <svg className="w-2 h-2 absolute top-0 right-0 m-4 pointer-events-none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 412 232"><path d="M206 171.144L42.678 7.822c-9.763-9.763-25.592-9.763-35.355 0-9.763 9.764-9.763 25.592 0 35.355l181 181c4.88 4.882 11.279 7.323 17.677 7.323s12.796-2.441 17.678-7.322l181-181c9.763-9.764 9.763-25.592 0-35.355-9.763-9.763-25.592-9.763-35.355 0L206 171.144z" fill="#648299" fill-rule="nonzero"/></svg>
+          <select onChange={handleSelectChange} className={ 
```

Comment:
for the select, you could add a value={selectedAPI} so it should the initial value on page load. Small detail but one that adds to the user experience.